pkg/util/helm_helpers.go

```diff
@@ -22,96 +22,308 @@ under the License.
 package util
 
 import (
+	"bytes"
 	"fmt"
-	"os/exec"
+	"net/http"
+	"path/filepath"
 	"regexp"
+	"strings"
 
 	log "github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/kube"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+	"k8s.io/helm/pkg/releaseutil"
+	"k8s.io/klog"
 )
 
-// RunHelm3 executes a helm command
-// It takes in a helm command, arguments to the command, and values to set in the helm chart
-func RunHelm3(commandName string, args []string, setValuesMap map[string]string) (string, error) {
-	var helmExists bool
-	var err error
-	if helmExists, err = HelmV3Exists(); err != nil {
-		return "", err
-	}
-	if !helmExists {
-		return "", fmt.Errorf("helm v3 is not installed in PATH")
-	}
-	cmdArgs := genHelm3Args(commandName, args, setValuesMap)
-	cmd := exec.Command("helm", cmdArgs...)
-	log.Debugf("%+v", cmd)
-	stdoutErr, err := cmd.CombinedOutput()
+var settings = cli.New()
+
+// CreateWithHelm3 uses the helm NewInstall action to create a resource in the cluster
+// Modified from https://github.com/openshift/console/blob/master/pkg/helm/actions/install_chart.go
+func CreateWithHelm3(releaseName, namespace, chartURL string, vals map[string]interface{}, kubeConfig string, dryRun bool) error {
```

Comment:
Only glanced over this file.  Let's just make sure to note that exact commit source of the copy/paste, so we can follow-along the git-jitsu 😄